File: VB/WindowsApplication1/WindowsApplication1/AdvancedXtraForm/AdvancedRibbonControl.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using System.Drawing;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class AdvancedRibbonControl : RibbonControl
    {

        public AdvancedRibbonControl()
        {
            Toolbar.ShowCustomizeItem = false;
            RibbonStyle = RibbonControlStyle.MacOffice;
        }

        protected override RibbonViewInfo CreateViewInfo()
        {
            return new AdvancedRibbonViewInfo(this);
        }
    }

    public class AdvancedRibbonViewInfo : RibbonViewInfo
    {
        public AdvancedRibbonViewInfo(RibbonControl ribbon)
            : base(ribbon)
        {
            
        }


        protected override int CalcMinHeight()
        {
            return Caption.CalcCaptionHeight();
        }

    }
}

File: VB/VB/WindowsApplication1/AdvancedXtraForm/AdvancedRibbonControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports System.Drawing

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory("")> _
	Public Class AdvancedRibbonControl
		Inherits RibbonControl

		Public Sub New()
			Toolbar.ShowCustomizeItem = False
			RibbonStyle = RibbonControlStyle.MacOffice
		End Sub

		Protected Overrides Function CreateViewInfo() As RibbonViewInfo
			Return New AdvancedRibbonViewInfo(Me)
		End Function
	End Class

	Public Class AdvancedRibbonViewInfo
		Inherits RibbonViewInfo
		Public Sub New(ByVal ribbon As RibbonControl)
			MyBase.New(ribbon)

		End Sub


		Protected Overrides Function CalcMinHeight() As Integer
			Return Caption.CalcCaptionHeight()
		End Function

	End Class
End Namespace
